cocotb/share/lib/gpi/GpiCommon.cpp

```diff
@@ -160,18 +162,37 @@ static void gpi_load_libs(std::vector<std::string> to_load)
          iter++)
     {
         void *lib_handle = NULL;
-        std::string full_name = "lib" + *iter + DOT_LIB_EXT;
+
+        std::string arg = *iter;
+
+        std::string lib_name;
+        std::string func_name;
+        auto it = std::find(arg.begin(), arg.end(), ':');
+        bool const has_colon = it == arg.end();
+        if (has_colon) {
+            // no colon in the string, default
+            lib_name = arg;
+            func_name = arg + "_entry_point";
+        }
+        else {
+            lib_name = std::string(arg.begin(), it);
+            func_name = std::string(it+1, arg.end());
+        }
+
+        std::string full_name = "lib" + lib_name + DOT_LIB_EXT;
         const char *now_loading = (full_name).c_str();
 
         lib_handle = utils_dyn_open(now_loading);
         if (!lib_handle) {
             printf("cocotb: Error loading shared library %s\n", now_loading);
```

Comment:
Is `now_loading` needed? It was replaced below with `full_name.c_str()` directly.

Reply:
Nice catch